shared/src/commonMain/kotlin/org/hyperskill/app/main/presentation/AppActionDispatcher.kt

```diff
@@ -67,10 +72,11 @@ internal class AppActionDispatcher(
         if (isSubscriptionPurchaseEnabled) {
             currentSubscriptionStateRepository
                 .changes
-                .map { it.type }
-                .distinctUntilChanged()
-                .onEach { subscriptionType ->
-                    onNewMessage(AppFeature.InternalMessage.SubscriptionTypeChanged(subscriptionType))
+                .distinctUntilChangedBy {
+                    Triple(it.type, it.status, it.validTill)
```

Comment:
For me it's unclear why we should observe only changes in `type`, `status` and `validTill` of the subscription, why not use `distinctUntilChanged()` on whole subscription model itself?

Reply:
I used filtration by `type`, `status`, and `validTill` fields to reduce sending the `SubscriptionChanged` message.
I agree, that it is redundant in this case.